src/main/java/net/sf/mzmine/modules/peaklistmethods/identification/formulapredictionpeaklist/FormulaPredictionPeakListTask.java

```diff
@@ -163,18 +163,44 @@ public void run() {
           massRange.upperEndpoint(), elementCounts);
 
       IMolecularFormula cdkFormula;
+
+      // create a map to store ResultFormula and relative mass deviation for sorting
+      Map<Double, ResultFormula> possibleFormulas = new HashMap<Double, ResultFormula>();
       while ((cdkFormula = generator.getNextFormula()) != null) {
+        if (isCanceled())
+          return;
+
         // Mass is ok, so test other constraints
-        checkConstraints(cdkFormula, row);
+        ResultFormula resultEntry = checkConstraints(cdkFormula, row);
+
+        // calc rel mass deviation
+        Double relMassDev = ((searchedMass

```

Comment:
It is a good practice to avoid too many function calls in one line, because when an exception occurs, you won't know which function call had caused it. Why not separate these into three different lines
- MolecularFormulaManipulator.getString(cdkFormula))
- FormulaUtils.calculateExactMass(...)
- relMassDev = ...

Reply:
agreed